https/tls_config_test.go

```diff
@@ -0,0 +1,357 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package https
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+)
+
+var (
+	logging bool
+	port    = ":5000"
+)
+
+type TestInputs struct {
+	Name           string
+	Server         *http.Server
+	YAMLConfigPath string
+	httpClient     *http.Client
+	Client         func() *http.Client
+	ConnectionURL  string
+	ExpectedResult bool
+}
+
+func TestListen(t *testing.T) {
```

Comment:
I would recommend that you look at https://golang.org/pkg/net/http/httptest/ and see how it can be used here rather than starting the HTTP server on a fixed port that may be already taken.

Reply:
httptest won't work with the tests as it's more for testing the handler as opposed to whether the server starts as intended. We've adjusted the function so that it finds an available port and runs on there but I think this is the reason it's failing the buildkite tests